### ISSUE TITLE
SyntaxEditor.SetAccessibility adds an extra newline trivia to declaration

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#2658]
```C#
using System;

public interface IGeneral
{
    object DoSomething();
}

public class ImplementsGeneralThree : IGeneral
{
    object IGeneral.DoSomething() { return null; }

    // private, this is no good
    int DoSomething() { Console.WriteLine(this); return 0; }
}
```

Invoke `SyntaxEditor.SetAccessibility` API for the private method DoSomething with the comment above. (API: http://source.roslyn.io/Microsoft.CodeAnalysis.Workspaces/R/b82c7b6de611d383.html)

I have the below code in the fixer:
```C#
        private async Task<Document> MakeProtected(Document document, ISymbol symbolToChange, CancellationToken cancellationToken)
        {
            var editor = SymbolEditor.Create(document);

            await editor.EditAllDeclarationsAsync(symbolToChange, (docEditor, declaration) =>
            {
                docEditor.SetAccessibility(declaration, Accessibility.Protected);
            }, cancellationToken);

            return editor.GetChangedDocuments().First();
        }
```

**Got:**
```C#
using System;

public interface IGeneral
{
    object DoSomething();
}

public class ImplementsGeneralThree : IGeneral
{
    object IGeneral.DoSomething() { return null; }


    // private, this is no good
    protected int DoSomething() { Console.WriteLine(this); return 0; }
}
```

NOTE the extra newline added before the comment "// private, this is no good" in the returned declaration.

**Comment by bkoelman** [comment]
Possibly related to #1947.

**Comment by mattwar** [comment]
SyntaxEditor doesn't add any trivia explicitly. It is more likely a formatting issue caused by marker trivia introduced with the new modifier token.

**Comment by mattwar** [comment]
Fixed with #2701

**Comment by lgolding** [comment]
Verified as fixed.
